### PR TITLE
Audit log report

### PR DESCRIPTION
**Pull request by dadarya0**
| Q                                      | A
| -------------------------------------- | ---
| Bug fix? (use the a.b branch)          | No
| New feature/enhancement? (use the a.x branch)      | Yes
| Deprecations?                          | [ ]
| BC breaks? (use the c.x branch)        | No
| Automated tests included?              | Yes 
| Related user documentation PR URL      | mautic/mautic-documentation#... 
| Related developer documentation PR URL | mautic/developer-documentation#... 
| Issue(s) addressed                     | Fixes #... 

#### Description:

This PR introduces a new report data source, Audit Log.
This will enable users to see and analyse audit log in a report format.
The audit log details are shown in prettified JSON format where 0 represents old and 1 represents new values.
This report allows only for admin user

#### Steps to test this PR:

1. Open this PR on Gitpod or pull down for testing locally (see docs on testing PRs [here](https://contribute.mautic.org/contributing-to-mautic/tester))
2. Create new report
3. Select Audit logs as Data source
4. Select columns and save report

### CODE REVIEW

**Review by aarohiprasad — DISMISSED** [review]
Looks good to me. Thanks @dadarya0

File: app/bundles/ReportBundle/Resources/views/Report/_details_report_content.html.twig [hunk 1]
```diff
@@ -81,10 +82,30 @@
...
                                             {% endif %}
                                             {% if closeLink %}</a>{% endif %}
+                                            {% if collapse %}
+                                            <div class="accordian-body collapse" id="audit-log-details-{{ startCount }}">
+                                                <pre id="pre-audit-log-details-{{ startCount }}"></pre>
+                                            </div>
+                                            <script>
+                                                var txt = document.createElement("textarea");
+                                                txt.innerHTML = '{{ (value) }}';
+                                                console.log(txt.value);
```
Comment (dadarya0) [inline app/bundles/ReportBundle/Resources/views/Report/_details_report_content.html.twig]:
```suggestion
```

**Review by aarohiprasad — DISMISSED** [review]

**Review by kuzmany — DISMISSED** [review]
Nice, works as expected 👍

**Comment by dadarya0** [comment]
@aarohiprasad @kuzmany could you please review again https://github.com/mautic/mautic/pull/13373/commits/cc0027778a911ca22837d52944571ae571cbaadc as sorting, searching, pagination everything was broken in 5.x and I have fixed same in this PR too.

**Review by aarohiprasad — APPROVED** [review]

**Comment by RCheesley** [comment]
@dadarya0 no docs PR was made for this PR. If you have chance can you please add one to add the necessary: https://docs.mautic.org/en/5.x/reports/reports.html#data-sources

cc @mautic/education-team as this is required for the 5.1 release.

**Pull request merged**